spec/services/mediator_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe Mediator do
+  subject { described_class }
+
+  let(:collection) { create(:collection) }
+  let(:user) { create(:user) }
+  let(:file) { nil }
+  let(:filename) { nil }
+  let(:link) { nil }
+  let(:chunking_method) { :basic }
+  let(:chunking_profile) { create(:chunking_profile, method: chunking_method, size: 800) }
+  let(:doc) { Document.new(state: :created, link:, file:, filename:, chunking_profile:, collection:, user:) }
```

Comment:
Try `create(:document, ...)`. That will save it for you, too.

Or, if you would prefer not to use the factory, just use `Document.create!` here.

Reply:
Earlier on I had a lot of trouble with `create(:document ...)` but now, I guess since I'm passing all parameters from the test, when I switched it works. Sheesh.